src/school/school.service.v2.ts

```
@@ -559,4 +562,111 @@ export class SchoolServiceV2 extends SchoolService {
+  
+        let nyay_panchayat_id;
+  
+        // Check for nypanchayat existence using Lodash's `_.isNil`
+        if (!_.isNil(nypanchayat)) {
+          const nypanchayatDetails = await this.prismaService.nyay_panchayats.findUnique({
+            where: { name_district_id_block_id: { block_id, district_id, name: nypanchayat } },
+          });
+          nyay_panchayat_id = _.get(nypanchayatDetails, 'id');
+        }
+  
+        // Create payload for creating/updating entries in DB
+        const payload: CreateSchoolListDto = {
+          ...element,
+          block_id,
+          district_id,
+          nyay_panchayat_id,
+        };
+  
+        try {
+          // Upsert using prisma and handle success/failure cases
+          const response: any = await this.prismaService.school_list.upsert({
+            where: { udise: payload.udise },
+            update: payload,
+            create: payload,
+          });
+          successSchoolList.push(response);
+        } catch (error) {
+          const { errorMessage } = getPrismaErrorStatusAndMessage(error);
+          failureSchoolList.push({ ...payload, message: errorMessage });
+        }
+      })
+    );
+  
+    return { successSchoolList, failureSchoolList };
+  }  
+
+  async parseExcel(buffer: Buffer): Promise<SchoolListExcelDataDto[]> {
+    // 1. Read the Excel file using XLSX and get the first sheet and data
+    const workbook = XLSX.read(buffer, { type: 'buffer' });
+    const sheetName = workbook.SheetNames[0];
+    const worksheet = workbook.Sheets[sheetName];
+    const data: any = XLSX.utils.sheet_to_json(worksheet, { header: 1 });
+
+    // 2. Leverage Lodash for efficient array and object manipulation
+    const parsedData = _.chain(data.slice(1)) // Use Lodash to chain array operations and start from the second row
+      .map((row: any) => {
+        const rowData: any = {};
+        _.forEach(data[0], (header: string, index: number) => {
```

data[0] or data.slice(1) and workbook.SheetNames[0] will cause internal error if CSV is not correct. You would that error to be propagated to the user instead of logging only to console.